### ISSUE TITLE
I would like InnerBlocks component to have the ability to "keepPlaceholderOnBlur"

### ISSUE THREAD

**Issue by jakeparis** [WordPress/gutenberg#35687]
## What problem does this address?
Currently when I have a block with an empty innerBlocks component, and the block loses focus, the innerBlocks component completely collapses. That causes everything below it to shift up. When performing a copy-paste job to update a page with a new bunch of these blocks, the constant up-and-down shifting of the content is a frustrating experience, and sometimes leads to selecting the wrong text.

![2021-10-15_13-47-58](https://user-images.githubusercontent.com/7408985/137531321-e141d74a-1637-4c13-a3f0-60a473a4be9a.gif)

## What is your proposed solution?
Provide the innerBlocks component with a `keepPlaceholderOnBlur` attribute, similar to the `keepPlaceholderOnFocus` attribute on the RichText component.

**Comment by stokesman** [comment]
Could this not be resolved by using a [`template`](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/#template)? Blocks in the template can have a placeholder and it won't disappear when focus is elsewhere.

**Comment by jakeparis** [comment]
@stokesman You are correct, adding a template with just an empty `core/paragraph` was enough to achieve the desired behavior: the block _looks_ empty, but it doesn't collapse. Thank you for that fix, and this is good enough for my needs here,  but I will say that it feels a bit like a bandaid, when an attribute would be a more explicit solution.

**Comment by talldan** [comment]
This is supposed to be already possible via `renderAppender` (the placeholder is a block appender) - docs here (not sure why there aren't in the handbook):
https://github.com/WordPress/gutenberg/blob/bd694faecb0061f420bbb67763ce7df07ae26ff4/packages/block-editor/src/components/inner-blocks/README.md#renderappender

So you'd use `InnerBlocks.DefaultBlockAppender` as the value to achieve what you want (if there wasn't the bug I mentioned below). If your block supports multiple block types then my recommendation would be the `InnerBlocks.ButtonBlockAppender` like the group block uses.

At the moment it looks like `InnerBlocks.DefaultBlockAppender` is broken, so I've made a separate bug report:
[Using InnerBlocks.DefaultBlockAppender for renderAppender is visually broken when the block is deselected](https://github.com/WordPress/gutenberg/issues/74060#top)

I'll close this issue seeing as the APIs already exist and there's now a bug report tracking the bug.
